controller/state/statedb.go

```diff
@@ -22,6 +21,15 @@ import (
 
 var log = logging.Logger("controller-state")
 
+type errorString string
+
+func (e errorString) Error() string {
+	return string(e)
+}
+
+const ErrNotAssigned = errorString("tasks must be acquired through pop task")
+const ErrWrongWorker = errorString("task already acquired by other worker")
```

Comment:
This is fine, but I was curious why do this above, and not
```go
var (
    ErrNotAssigned = errors.New("tasks must be acquired through pop task")
    ErrWrongWorker = errors.New("task already acquired by other worker")
)
```
as this allows you to compare `if err == ErrNotAssigned { ...`

Reply:
that will still work. (arguably it's a slightly stricter form of equality but that also is good I think)

the main difference here it's a way to actually make errors constant. It drives me nuts that any program anywhere can reassign the values of sentinel errors in go.

Reply:
see: https://dave.cheney.net/tag/errors